doc/persistence.rst

```diff
@@ -0,0 +1,103 @@
+..
+    For doctests:
+
+    >>> from joblib.testing import warnings_to_stdout
+    >>> warnings_to_stdout()
+
+.. _persistence:
+
+===========
+Persistence
+===========
+
+.. currentmodule:: joblib.numpy_pickle
+
+Usecase
+=======
+
+:func:`joblib.dump` and :func:`joblib.load` provide a replacement for
+pickle to work efficiently on Python objects containing large data, in
+particular large numpy arrays.
+
+A simple example
+================
+
+  First we create a temporary directory:
```

Comment:
why the indentation before "First"?

Also I would put `::` instead of `:` and a blank line before the code block.